lib/services/security/checkpoint.js

```diff
@@ -213,7 +213,9 @@ CheckPoint.prototype.__loadPermissionSet = function(identity, callback) {
           if (e) return eachCB(e);
           if (group == null) return eachCB();
           for (let permissionPath in group.permissions)
-            permissions[permissionPath] = group.permissions[permissionPath];
+            permissions[permissionPath] = this.utilsService.clone(

```

Comment:
Why are we cloning here?

Reply:
Because we later adjust the perrmissions[permissionPath] object (Line 226-229), this can result in changes to the cached group permissions which is a problem. I  don't think there's a _defaultsDeepWith available.

Reply:
I may be able to find another way around this, will look tomorrow

Reply:
Yeah, this needs to stay, unfortunately - we have to clone either here or when we get the permission set out of the cache..